glib/src/main_context.rs

```diff
@@ -124,6 +124,29 @@ impl MainContext {
         let _thread_default = ThreadDefaultContext::new(self);
         func()
     }
+
+    #[doc(alias = "g_main_context_acquire")]
+    pub fn acquire(&self) -> Result<MainContextAcquireGuard, crate::BoolError> {
+        unsafe {
+            let ret: bool = from_glib(ffi::g_main_context_acquire(self.to_glib_none().0));
+            if ret {
+                Ok(MainContextAcquireGuard(self))
+            } else {
+                Err(bool_error!("Failed to acquire main context"))
+            }
+        }
+    }
+}
+
+#[must_use = "if unused the main context will be released immediately"]
+pub struct MainContextAcquireGuard<'a>(&'a MainContext);
+
+impl<'a> Drop for MainContextAcquireGuard<'a> {
+    fn drop(&mut self) {
```

Comment:
Could use a doc alias to help people finding out about the Drop implementation maybe?

Reply:
Doc alias for which function?

Reply:
for drop, this way the user can find it by searching for the C function

Reply:
For `g_main_context_release` you mean?

Reply:
Yes

Reply:
Ok, no idea if that actually has any effect though. Probably not :)

Reply:
It does https://gtk-rs.org/gtk4-rs/git/docs/?search=gdk_event_get_type :)